### ISSUE TITLE
Improving YOLOv5 Inference Speed on CPU for Detection

### ISSUE THREAD

**Issue by gauricollab09** [ultralytics/yolov5#13630]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi everyone,

I'm using YOLOv5 for a logo detection. On GPU (RTX A6000), the inference speed is excellent : around 30+ FPS. However, when running on CPU (a reasonably powerful machine), the inference speed drops significantly to about 1 frame every 2 seconds (~0.5 FPS), which is too slow.

Is there a way to speed this up on CPU? Even achieving 8–9 FPS would be a huge improvement. Are there any flags, quantization techniques or runtime options you recommend?

Any suggestions if you could give would be useful. Thanks in advance!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @gauricollab09, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) for guidance, including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and advanced topics like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** (MRE) so we can help debug more efficiently.

If this is a custom training or inference ❓ Question (such as yours about CPU speed), please provide as much detail as possible, including:
- Model version and export format (e.g., PyTorch, ONNX, etc.)
- Example command(s) used for inference
- Specs of your CPU environment
- Any logs or error messages

You can also review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) for general optimization advice.

## Requirements
[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed, including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```
## Environments
YOLOv5 runs in any of these up-to-date verified environments (with all dependencies, including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

_This is an automated response. An Ultralytics engineer will also assist you here soon._ 😊

**Comment by pderrenger** [comment]
For significant CPU inference improvements, I'd recommend trying these optimization approaches in order of expected impact:

First, export your model to ONNX format and use optimized inference engines like [OpenVINO for Intel hardware](https://docs.ultralytics.com/integrations/openvino/) which can provide substantial speedups on CPU. You can also try [DeepSparse with Neural Magic's sparsified YOLOv5 models](https://www.ultralytics.com/blog/deploy-yolov5-with-neural-magics-deepsparse-for-gpu-class-performance-on-cpus) which achieves 2.8x-5.8x performance gains over standard ONNX Runtime on CPU.

Additionally, ensure you're using batch size 1 for lowest latency, consider model quantization (INT8), and if accuracy allows, try YOLOv5n which is specifically designed for lightweight deployments. The reference material shows OpenVINO achieving 23ms inference times on Intel i9 compared to 115ms with standard PyTorch on CPU.
